ui/apps/platform/src/services/CollectionsService.ts

```diff
@@ -81,6 +70,87 @@ export type ResolvedCollectionResponse = {
+    'Namespace Annotation': 'Namespace',
+    Cluster: 'Cluster',
+    'Cluster Label': 'Cluster',
+    'Cluster Annotation': 'Cluster',
+};
+
+function isAggregateError(value: Collection | AggregateError): value is AggregateError {
+    return value instanceof AggregateError;
+}
+
+/**
+ * This function takes a raw `CollectionResponse` from the server and parses it into a representation
+ * of a `Collection` that can be supported by the current UI controls. If any incompatibilities are detected
+ * it will return a list of validation errors to the caller.
+ */
+function parseCollection(data: CollectionResponse): Collection | AggregateError {
+    const collection: Collection = {
+        name: data.name,
+        description: data.description,
+        embeddedCollectionIds: data.embeddedCollections.map(({ id }) => id),
+        selectorRules: {
+            Deployment: {},
+            Namespace: {},
+            Cluster: {},
+        },
+    };
+
+    const errors: string[] = [];
+
+    if (data.resourceSelectors.length > 1) {
+        errors.push(
+            `Multiple 'ResourceSelectors' were found for this collection. Only a single resource selector is supported in the UI. Further validation errors will only apply to the first resource selector in the response.`
+        );
+    }
+
+    data.resourceSelectors[0]?.rules.forEach((rule) => {
+        const entity = fieldToEntityMap[rule.fieldName];
+        const field = rule.fieldName;
+        collection.selectorRules[entity] = collection.selectorRules[entity] ?? {
+            field,
+            rules: [],
+        };
+
+        const hasMultipleFieldsForEntity = collection.selectorRules[entity].field !== field;
+        const isRuleForAnnotationField = rule.fieldName.endsWith('Annotation');
+        const isUnsupportedRuleOperator = rule.operator !== 'OR';
+
+        if (hasMultipleFieldsForEntity) {
+            errors.push(
+                `Each entity type can only contain rules for a single field. A new rule was found for [${entity} -> ${field}], when rules have already been applied for [${entity} -> ${collection.selectorRules[entity].field}].`
+            );
+        }
+        if (isRuleForAnnotationField) {
+            errors.push(
+                `Collection rules for 'Annotation' field names are not supported at this time. Found field name [${rule.fieldName}].`
+            );
+        }
+        if (isUnsupportedRuleOperator) {
+            errors.push(
+                `Only the disjunction operation ('OR') is currently supported in the front end collection editor. Received an operator of [${rule.operator}].`
+            );
+        }
+
+        if (
+            !hasMultipleFieldsForEntity &&
+            !isRuleForAnnotationField &&
+            !isUnsupportedRuleOperator
+        ) {
+            collection.selectorRules[entity].rules.push(rule);
+        }
+    });
+
+    return errors.length > 0 ? new AggregateError(errors) : collection;
```

Comment:
If it isn't useful to have the full list of validation errors, I can simplify this code to a generic single error and remove the `AggregateError` return type.

Reply:
Although I will let this simmer in my slow cooker today, initial impression is tight coupling for service to import types from container and for service to significantly transform response from request.

Reply:
What would it look like if:
* service type and return value corresponds to backend behavior (and preview in network panel)
* container has limited types and pure util functions to transform data and describe errors

Reply:
I went back and forth on this, and originally had the transform functions right in the component file but changed it yesterday.

I'm OK with moving the transform functions and types out of the service and back into the container. I can see there being a use case where we want to get the full server response via the service function and use the result someplace else, someplace that might support the full structure beyond MVP.

Reply:
For example, here is a surprise. A teammate from long ago called it asymetritis.
* policies container has transformation functions from response to form state and vice versa
* frontend policies service function for POST and PUT requests also call a util function